romeo/src/state.rs

```diff
@@ -0,0 +1,218 @@
+use bdk::bitcoin::{
+    Address as BitcoinAddress, Block, Transaction as BitcoinTransaction, Txid as BitcoinTxId,
+};
+use blockstack_lib::burnchains::Txid as StacksTxId;
+use blockstack_lib::chainstate::stacks::StacksTransaction;
+use blockstack_lib::types::chainstate::StacksAddress;
+use tracing::debug;
+
+use crate::config::Config;
+use crate::event::Event;
+use crate::event::TransactionStatus;
+use crate::task::Task;
+
+/// The whole state of the application
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct State {
+    contract: Option<Contract>,
+    deposits: Vec<Deposit>,
+    withdrawals: Vec<Withdrawal>,
+    next_stx_nonce: u64,
+    block_height: u64,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
```

Comment:
Is it of particular importance to derive `serde::Serialize, serde::Deserialize` instead of just `Serialize, Deserialize` like in the other files?

Reply:
Done :heavy_check_mark: